tests/framework/framework.go

```
@@ -126,6 +135,53 @@ func NewInforming(ctx context.Context, initialObjects ...runtime.Object) *Framew
+// NewForConfig creates a new test framework backed by real clients from a
+// rest.Config. This is intended for envtest-style tests where a real
+// kube-apiserver is available and CRDs are installed externally (e.g. via
+// envtest.Environment.CRDDirectoryPaths).
+//
+// Unlike NewInforming, no CRD informer is started; GVK-to-resource resolution
+// uses the discovery-backed REST mapper instead.
+//
+// Options are initialised with production defaults without calling
+// options.Read() (which registers flags that conflict with controller-runtime
+// imports). Ports are left unset and allocated by Start().
+func NewForConfig(ctx context.Context, cfg *rest.Config) (*Framework, error) {
```

golangci-lint (2.12.2)

[warning] unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _

(revive)